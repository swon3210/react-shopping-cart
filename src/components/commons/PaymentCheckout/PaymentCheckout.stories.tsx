import PaymentCheckout, { Props } from './PaymentCheckout';

export default {
  title: 'Components/Commons/PaymentCheckout',
  component: PaymentCheckout,
  argTypes: {},
};

const Template = (args: Props) => <PaymentCheckout {...args} />;

export const Default = Template.bind({});

(Default as any).args = {
  title: '결제금액',
  priceLabel: '총 결제금액',
  price: '18,900원',
  buttonText: '주문하기(2개)',
};
