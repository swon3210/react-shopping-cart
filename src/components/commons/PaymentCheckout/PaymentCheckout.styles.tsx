import styled from 'styled-components';

export const PaymentCheckout = styled.div`
  display: inline-block;
  border: 1px solid ${({ theme }) => theme.GRAY_200};
  color: ${({ theme }) => theme.TEXT_COLOR};
`;

export const Title = styled.div`
  padding: 22px 30px;
  font-size: 24px;
  border-bottom: 3px solid ${({ theme }) => theme.GRAY_200};
`;

export const Container = styled.div`
  padding: 35px 30px;
`;

export const PriceWrapper = styled.div`
  display: flex;
  justify-content: space-between;
  margin-bottom: 66px;
`;

export const PriceLabel = styled.span`
  position: relative;
  font-size: 20px;
  font-weight: bold;
  &::after {
    content: '';
    position: absolute;
    left: 0;
    bottom: 0;
    width: 100%;
    height: 6px;
    background-color: ${({ theme }) => theme.PRIMARY_COLOR};
    opacity: 0.4;
  }
`;

export const Price = styled.span`
  position: relative;
  font-size: 20px;
  font-weight: bold;
  &::after {
    content: '';
    position: absolute;
    left: 0;
    bottom: 0;
    width: 100%;
    height: 6px;
    background-color: ${({ theme }) => theme.PRIMARY_COLOR};
    opacity: 0.4;
  }
`;
