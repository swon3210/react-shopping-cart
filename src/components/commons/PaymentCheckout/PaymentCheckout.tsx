import Button from '../Button/Button';
import * as Styled from './PaymentCheckout.styles';

export interface Props {
  title: string;
  priceLabel: string;
  price: string;
  buttonText: string;
  isButtonDisabled?: boolean;
  onButtonClick?: () => void;
}

const PaymentCheckout = ({ title, priceLabel, price, buttonText, onButtonClick, isButtonDisabled = false }: Props) => {
  return (
    <Styled.PaymentCheckout>
      <Styled.Title>{title}</Styled.Title>
      <Styled.Container>
        <Styled.PriceWrapper>
          <Styled.PriceLabel>{priceLabel}</Styled.PriceLabel>
          <Styled.Price>{price}</Styled.Price>
        </Styled.PriceWrapper>
        <Button size="MD" onClick={onButtonClick} disabled={isButtonDisabled}>
          {buttonText}
        </Button>
      </Styled.Container>
    </Styled.PaymentCheckout>
  );
};

export default PaymentCheckout;
