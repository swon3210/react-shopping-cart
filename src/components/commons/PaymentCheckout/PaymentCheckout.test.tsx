import PaymentCheckout from './PaymentCheckout';
import { render } from '@testing-library/react';

describe('PaymentCheckout Component', () => {
  it('PaymentCheckout Snapshot', () => {
    const paymentCheckoutUtil = render(
      <PaymentCheckout buttonText="buttonText" price={'2000'} title="title" priceLabel="priceLabel" />
    );
    expect(paymentCheckoutUtil.asFragment()).toMatchSnapshot();
  });
});
