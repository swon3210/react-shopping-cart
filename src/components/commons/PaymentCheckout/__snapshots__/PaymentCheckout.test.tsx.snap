// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PaymentCheckout Component PaymentCheckout Snapshot 1`] = `
<DocumentFragment>
  <div
    class="sc-gsDJrp gCAlHF"
  >
    <div
      class="sc-dkPtyc gtzEzS"
    >
      title
    </div>
    <div
      class="sc-hKwCoD jcydiL"
    >
      <div
        class="sc-eCImvq bnNPYq"
      >
        <span
          class="sc-jRQAMF FLdNx"
        >
          priceLabel
        </span>
        <span
          class="sc-gKckTs iDFVdh"
        >
          2000
        </span>
      </div>
      <button
        class="sc-bdvvaa jvAQYl"
        type="button"
      >
        buttonText
      </button>
    </div>
  </div>
</DocumentFragment>
`;
